cylc/uiserver/data_store_mgr.py

```diff
@@ -67,40 +71,98 @@ def __init__(self, workflows_mgr):
         self.executors = {}
         self.delta_queues = {}
 
-    async def sync_workflow(self, w_id, *args, **kwargs):
+    def update_contact(self, w_id, contact_data=None):
+        delta = DELTAS_MAP[WORKFLOW]()
+        delta.time = time.time()
+        flow = delta.updated
+        flow.id = w_id
+        if contact_data:
+            # update with contact file data
+            flow.name = contact_data['name']
+            flow.owner = contact_data['owner']
+            flow.host = contact_data[CFF.HOST]
+            flow.port = int(contact_data[CFF.PORT])
+            # flow.pub_port = int(contact_data[CFF.PUBLISH_PORT])
```

Comment:
Why port and not publish port?  Maybe we'll go back to only having one when we do event driven response (?)

Reply:
I commented it out because it caused traceback, looks like `pub_port` [isn't in the proto schema](https://github.com/cylc/cylc-flow/blob/fed43a149b76ff5033d4b19eb26fb287936ab8f3/cylc/flow/data_messages.proto#L72).

Reply:
Ah yeah, opps!.. That's in my PR:
https://github.com/cylc/cylc-flow/pull/3757/files#diff-ba2245097b83947ce98caf52bdbef654

(I tested it against that)